spec/models/manageiq/providers/hawkular/middleware_manager/middleware_server_spec.rb

```diff
@@ -0,0 +1,64 @@
+describe ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer do
+  let(:ems_hawkular) do
+    allow(MiqServer).to receive(:my_zone).and_return("default")
+    auth = AuthToken.new(:name => "test", :auth_key => "valid-token", :userid => "jdoe", :password => "password")
+    FactoryGirl.create(:ems_hawkular,
+                       :hostname        => 'localhost',
+                       :port            => 8080,
+                       :authentications => [auth])
+  end
+
+  it "will query live metrics for all metrics available" do
+    end_time = Time.new(2016, 3, 18, 0, 0, 0, "+01:00")      # Fixed time for testing
+    start_time = Time.new(2016, 3, 17, 0, 0, 0, "+01:00")    # Fixed time for testing
+    interval = 20                                            # Interval in seconds
+    VCR.use_cassette(described_class.name.underscore.to_s) do # , :record => :new_episodes) do
+      EmsRefresh.refresh(ems_hawkular)
```

Comment:
You should not need to refresh all of hawkular just to collect metrics.  I would think you could create a middleware server directly in the database with hardcoded values for the server id or whatever is needed.  The metrics themselves though I would expect to be recorded.